docs/source/en/model_doc/pegasus.md

```diff
@@ -14,95 +14,115 @@ rendered properly in your Markdown viewer.
-*pegasus-large*, whence the other checkpoints are fine-tuned:
+The example below demonstrates how to summarize text with [`Pipeline`], [`AutoModel`], and from the command line.
 
-- Each checkpoint is 2.2 GB on disk and 568M parameters.
-- FP16 is not supported (help/ideas on this appreciated!).
-- Summarizing xsum in fp32 takes about 400ms/sample, with default parameters on a v100 GPU.
-- Full replication results and correctly pre-processed data can be found in this [Issue](https://github.com/huggingface/transformers/issues/6844#issue-689259666).
-- [Distilled checkpoints](https://huggingface.co/models?search=distill-pegasus) are described in this [paper](https://arxiv.org/abs/2010.13002).
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-## Implementation Notes
+```py
+import torch
+from transformers import pipeline
 
-- All models are transformer encoder-decoders with 16 layers in each component.
-- The implementation is completely inherited from [`BartForConditionalGeneration`]
-- Some key configuration differences:
-  - static, sinusoidal position embeddings
-  - the model starts generating with pad_token_id (which has 0 token_embedding) as the prefix.
-  - more beams are used (`num_beams=8`)
-- All pretrained pegasus checkpoints are the same besides three attributes: `tokenizer.model_max_length` (maximum
-  input size), `max_length` (the maximum number of tokens to generate) and `length_penalty`.
-- The code to convert checkpoints trained in the author's [repo](https://github.com/google-research/pegasus) can be
-  found in `convert_pegasus_tf_to_pytorch.py`.
+pipeline = pipeline(
+    task="summarization",
+    model="google/pegasus-xsum",
+    torch_dtype=torch.float16,
+    device=0
+)
+pipeline("""Plants are remarkable organisms that produce their own food using a method called photosynthesis.
+This process involves converting sunlight, carbon dioxide, and water into glucose, which provides energy for growth.
+Plants play a crucial role in sustaining life on Earth by generating oxygen and serving as the foundation of most ecosystems.""")
+```
 
-## Usage Example
+</hfoption>
+<hfoption id="AutoModel">
+
+```py
+import torch
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
+
+tokenizer = AutoTokenizer.from_pretrained(
+    "google/pegasus-xsum"
+)
+model = AutoModelForSeq2SeqLM.from_pretrained(
+    "google/pegasus-xsum",
+    torch_dtype=torch.float16,
+    device_map="auto",
+)
+
+input_text = """Plants are remarkable organisms that produce their own food using a method called photosynthesis.
+This process involves converting sunlight, carbon dioxide, and water into glucose, which provides energy for growth.
+Plants play a crucial role in sustaining life on Earth by generating oxygen and serving as the foundation of most ecosystems."""
+input_ids = tokenizer(input_text, return_tensors="pt").to("cuda")
+
+output = model.generate(**input_ids)
```

Comment:
```suggestion
output = model.generate(**input_ids, cache_implementation="static")
```

Reply:
Hmm, what version of Transformers are you running? It works for me on the latest version